strapi/strapi
# Add --cache option to lint and prettier scripts

### What does it do?

Adds the `--cache` and `--cache-strategy content` options to eslint and prettier scripts

### Why is it needed?

Linting currently takes a very long time (~50s on my local environment on `main`, ~80s when adding typescript linting in `features/deits`).

This change reduces subsequent runs after the first to a tiny fraction of the full run and should save developers about a minute of waiting before every strapi commit, since we run prettier and `lint --fix` with a commit hook.

### How to test it?

Run `yarn lint` and it should be cached and run faster on subsequent runs. When you update a file, it should be linted again the next time it's run.

The important thing is that it does not accidentally use the cache for something that should be re-checked. For that reason, I opted to use cache-strategy content in case switching between git branches could update content without updating a timestamp.

### Related issue(s)/PR(s)
